src/modules/launcher/PowerLauncher/SettingsWatcher.cs

```diff
@@ -19,28 +19,22 @@ namespace PowerLauncher
     public class SettingsWatcher : BaseModel
     {
         private static int MAX_RETRIES = 10;
+        private static object _watcherSyncObject = new object();
         private FileSystemWatcher _watcher;
         private Settings _settings;
         public SettingsWatcher(Settings settings)
         {
             _settings = settings;
             // Set up watcher
-            try
-            {
-                _watcher = Helper.GetFileWatcher(PowerLauncherSettings.POWERTOYNAME, "settings.json", OverloadSettings);
-            }
-            catch (Exception e)
-            {
-                Debug.WriteLine(e.Message);
-            }
+             _watcher = Helper.GetFileWatcher(PowerLauncherSettings.POWERTOYNAME, "settings.json", OverloadSettings);
 
             // Load initial settings file
             OverloadSettings();
         }
 
         public void OverloadSettings()
         {
-            Monitor.Enter(_watcher);
+            Monitor.Enter(_watcherSyncObject);

```

Comment:
I might be missing something, but where is `_watcherSyncObject` set and how is it different from `_watcher`?

Reply:
Hi @Alekhya.  It's initialized at the top of the file -- `private static object _watcherSyncObject = new object();`. 

This was done to fix [ca2002](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2002?view=vs-2019).   The `object` class does not have a weak identity and am therefore using it to lock the critical section instead of the filestystem watcher. 

> I might be missing something, but where is `_watcherSyncObject` set and how is it different from `_watcher`?

Hi @alekhyareddy28 . It's initialized at the top of the file -- private static object _watcherSyncObject = new object();.

This was done to fix ca2002. The object class does not have a weak identity and am therefore using it to lock the critical section instead of the filestystem watcher.

Reply:
Sounds good! Thanks for the clarification on locking weak identities.